templates/workloads/partials/cf/subscribe.yml

```
@@ -0,0 +1,226 @@
+    KmsMasterKeyId: !Ref SQSQueueKMSKey
+    {{- else}}
+    KmsMasterKeyId: 'alias/aws/sqs'
+    {{- end}}
+    {{- if .Subscribe.Queue.FIFO}}
+    FifoQueue: true
+    {{- if .Subscribe.Queue.FIFO.HighThroughput}}
+    FifoThroughputLimit: 'perMessageGroupId'
+    DeduplicationScope: 'messageGroup'
+    {{- end}}
+    {{- end}}
+    {{- if .Subscribe.Queue.DeadLetter}}
+    RedrivePolicy:
+      {{- if .Subscribe.Queue.DeadLetter.Id}}
+      deadLetterTargetArn: !Sub 'arn:${AWS::Partition}:sqs:${AWS::Region}:${AWS::AccountId}:{{logicalIDSafe .Subscribe.Queue.DeadLetter.Id}}'
+      {{- else}}
+      deadLetterTargetArn: !GetAtt DeadLetter.Arn
+      {{- end}}
+      maxReceiveCount: {{.Subscribe.Queue.DeadLetter.Tries}}
+    {{- end}}
+  {{- end}}
+
+{{- if .Subscribe.Queue}}{{- if .Subscribe.Queue.DeadLetter}}{{- if not .Subscribe.Queue.DeadLetter.Id}}
+DeadLetter:
+  Type: AWS::SQS::Queue
+  Properties:
+    QueueName: !Join ['', [!GetAtt SQSQueue.Name, "-", "DeadLetter"]]
+
+DeadLetterPolicy:
+  Type: AWS::SQS::QueuePolicy
+  Properties:
+    Queues: [!Ref 'DeadLetter']
+    PolicyDocument:
+      Version: "2012-10-17"
+      Statement:
+        - Sid: Allow-TaskRole-Subscribe
+          Effect: Allow
+          Principal:
+            Service: sns.amazonaws.com
+          Action: 
+            - sqs:GetMessage
+            - sqs:DeleteMessage
+          Resource: !GetAtt SQSQueue.Arn
+          Condition:
+            ArnEquals:
+              aws:SourceArn: !Ref 'TaskRole'
+{{- end}}{{- end}}{{- end}}
+
+{{- if .Subscribe.Queue}}{{- if .Subscribe.Queue.KMS}}
+SQSQueueKMSKey:
+  Type: AWS::KMS::Key
+  Properties:
+    KeyPolicy:
+      Version: '2012-10-17'
+      Statement:
+        - Effect: Allow
+          Principal:
+            AWS: !Sub 'arn:${AWS::Partition}:iam::${AWS::AccountId}:root'
+          Action: kms:*
+          Resource: '*'
+{{- end}}{{- end}}
+
+QueuePolicy:
+  Type: AWS::SQS::QueuePolicy
+  Properties:
+    Queues: [!Ref 'SQSQueue']
+    PolicyDocument:
+      Version: '2012-10-17'
+      Statement:
+        - Sid: Allow-TaskRole-Subscribe
+          Effect: Allow
+          Principal:
+            Service: sns.amazonaws.com
+          Action: 
+            - sqs:GetMessage
+            - sqs:DeleteMessage
+          Resource: !GetAtt SQSQueue.Arn
+          Condition:
+            ArnEquals:
+              aws:SourceArn: !Ref 'TaskRole'
+        {{- range $topic := .Subscribe.Topics}}
+        {{- if not $topic.Queue}}
+        - Effect: Allow
+          Principal:
+            Service: sns.amazonaws.com
+          Action: 
+            - sqs:SendMessage
+          Resource: !GetAtt SQSQueue.Arn
+          Condition:
+            ArnEquals:
+              aws:SourceArn: !Sub 'arn:${AWS::Partition}:sns:${AWS::Region}:${AWS::AccountId}:${AWS::StackName}-{{logicalIDSafe $topic.Name}}'
+        {{- end}}
+        {{- end}}
+
+{{- range $topic := .Subscribe.Topics}}
+{{logicalIDSafe $topic.Name}}SNSTopicSubscription:
```

It's possible that two different services, let's say `order` and `inventory` define the same topic name `product`. Can we prefix this with:
```suggestion
{{$topic.Service}}{{logicalIDSafe $topic.Name}}SNSTopicSubscription:
```